### DISCUSSION TITLE
Issue with Singularity CE 4.1.2-jammy and CASA-Distro 3.1.1 in Converting Image to Docker

### DISCUSSION THREAD

**Discussion by caro0031** [brainvisa/brainvisa.github.io#169] [Q&A] [answered]
Hello BrainVISA Support Team and Community, 

I am currently facing an issue while trying to convert a Singularity image file to a Docker container using the casa_distro_admin tool. My ultimate goal is to run BrainVISA scripts in batch mode through a Docker container, which is critical for integrating with our existing pipelines. 

Configuration Details: 
    Singularity Version: Singularity-CE 4.1.2-jammy 
    CASA-Distro Version: 3.1.1 
    Brainvisa Version : brainvisa-5.1.1.sif 

When attempting to convert a Singularity image without specifying the convert_from option, which should automatically detect the container type from the source metadata, I encounter the following error: ValueError: Unsupported converted container type: None 
According to the CASA-Distro documentation, the convert_from option should default to detecting the container type from the image metadata, yet it seems unable to do so, necessitating manual specification (convert_from=singularity). Thanks to this specification, my process went further but I faced another error concerning the metadata. 

Detailed Error: 

Traceback (most recent call last): 
  File "/home/user/brainvisa-5.1.1/casa-distro/bin/casa_distro_admin", line 25, in <module> 
    main() 
  File "/home/user/brainvisa-5.1.1/casa-distro/python/casa_distro/command.py", line 360, in main 
    result = command(*args, **kwargs) 
  File "/home/user/brainvisa-5.1.1/casa-distro/python/casa_distro/admin_commands.py", line 1509, in convert_image 
    image_id, msg = module.convert_image(base, metadata, output, 
  File "/home/user/brainvisa-5.1.1/casa-distro/python/casa_distro/docker.py", line 530, in convert_singularity_image_to_docker 
    image_tag = metadata['name'] 
KeyError: 'name' 

Are there known compatibility issues between Singularity CE 4.1.2-jammy and CASA-Distro 3.1.1 that might affect metadata recognition and handling? 
Is there a recommended approach or workaround for successfully converting Singularity images to Docker for use in BrainVISA, specifically for batch mode execution? 

Thank you very much for your time and help.

Carole-Anne

**Comment by c-langlet** [comment, marked as answer]
Hello,

Do you have the metadata json file in the same repository than your singularity image ?
If not, you can download it here : https://brainvisa.info/download/brainvisa-5.1.1.sif.json

**Reply by caro0031** [reply]
Hello,

I had a json named brainvisa-5.1.1.docker.json but I didn't have the sif one.
It solved my problem, I could build and run my docker created.

Thank you very much for your answer !
